server.js

```diff
@@ -2,22 +2,86 @@ const express = require("express");
 const cors = require("cors");
 
 const app = express();
+app.use(express.urlencoded({ extended: true }));
 
 app.use(cors());
+app.use(express.json());
+
+app.get("/", function (request, response) {
+	response.sendFile(__dirname + "./index.html");
+});
 
 const welcomeMessage = {
-  id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
+	id: 0,
+	from: "Bart",
+	text: "Welcome to CYF chat system!",
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
+
 const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
-});
+app.get('/search', (req, res) => {
+	const filteredMessages = messages.filter(mess => mess.text.toLowerCase().includes(req.query.term.toLowerCase()));
+	if (filteredMessages.length > 0) {
+		res.status(200).send(filteredMessages);
+	} else {
+		res.status(400).send('Nothing found');
+	}
+	console.log(req.query)
+})
+
+app.post('/messages', (req, res) => {
+	if ((req.body.from).length >= 3 && (req.body.text).length > 0) {
```

Comment:
Why did you choose `3` as the minimum for the length of the from field?

It you wanted to stick with `3`, you should move it into a constant to avoid having a magic number in your code
https://refactoring.guru/replace-magic-number-with-symbolic-constant

Reply:
My reasoning was that names should be at least 3 char long